snooty/parser.py

```diff
@@ -854,6 +862,180 @@ def handle_method_option(self, node: n.Directive) -> None:
         if target_idx >= 0:
             node.children.insert(0, node.children.pop(target_idx))
 
+    def handle_composable(self, node: n.ComposableDirective) -> None:
+        """Handles composable directive(s) and its children composable content. Translates string options to lists for consumption"""
+
+        # first convert specified options from str -> List[str]
+        option_ids_as_string = (
+            node.options["options"] if "options" in node.options else ""
+        )
+        default_ids_as_string = (
+            node.options["defaults"] if "defaults" in node.options else ""
+        )
+        option_ids: List[str] = re.split(r"\s*,\s*", option_ids_as_string.strip())
+        default_ids: List[str] = re.split(r"\s*,\s*", default_ids_as_string.strip())
+
+        # expect at least 1 option_ids
+        if len(option_ids) < 1:
+            self.diagnostics.append(
+                InvalidChildCount(
+                    "composable-tutorial", "option_ids", "at least one", node.start[0]
+                )
+            )
+
+        # get the expected composable options from the spec
+        spec_composables = specparser.Spec.get().composables
+        spec_composables_dict = {
+            expected_option.id: expected_option for expected_option in spec_composables
+        }
+
+        # validate the specified :options: and :defaults:
+        used_ids: Set[str] = set()
+        composable_options = []
+        default_ids_dict: Dict[str, str] = {}
+        for index in range(len(option_ids)):
+            option_id = option_ids[index]
+            try:
+                self.check_valid_option_id(
+                    option_id, node, spec_composables_dict, used_ids
+                )
+                composable_from_spec = next(
+                    (option for option in spec_composables if option.id == option_id),
+                )
+                specified_default_id = default_ids[index]
+                allowed_values_dict = {
+                    option.id: option for option in composable_from_spec.options
+                }
+                self.check_valid_option_id(
+                    specified_default_id, node, allowed_values_dict, set()
+                )
+                default_ids_dict[option_id] = (
+                    specified_default_id
+                    or composable_from_spec.default
+                    or composable_from_spec.options[0].id
+                )
```

Comment:
Same here as above: Maybe on 915 we can add ` if composable_from_spec.options else None` or `if composable_from_spec.options[0] else None` (unsure which is best) and then after check `if default_ids_dict[option_id] is None` and raise a specific Diagnostic?

Reply:
i did it for best practice, but essentially should not be hitting that line since we are already checking beforehand in the `check_valid_option_id` calls. also noting the tests didnt double down on the errors for invalid options :)